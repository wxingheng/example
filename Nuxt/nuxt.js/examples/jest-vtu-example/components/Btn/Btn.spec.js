import { shallowMount } from '@vue/test-utils'
import Btn from './Btn'

const factory = () => shallowMount(Btn, {
  propsData: {
    label: 'click me!'
  }
})

describe('Btn', () => {
  test('mounts properly', () => {
    const wrapper = factory()
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders properly', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('calls handleClick on click', () => {
    const wrapper = factory()
    const handleClickMock = jest.fn()
    wrapper.setMethods({
      handleClick: handleClickMock
    })
    wrapper.find('button').trigger('click')
    expect(handleClickMock).toHaveBeenCalled()
  })

  test('clicked is true after click', () => {
    const wrapper = factory()
    wrapper.find('button').trigger('click')
    expect(wrapper.vm.clicked).toBe(true)
  })
})
