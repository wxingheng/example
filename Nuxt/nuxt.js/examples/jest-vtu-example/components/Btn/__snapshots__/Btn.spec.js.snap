// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn renders properly 1`] = `
<button>
  click me!
</button>
`;
